### PR TITLE
Feature/RIFEX-84/Refactor-Domains-detail-page

### PR DESCRIPTION
**Pull request by rorolopetegui**
This is a refactor for the domain detail active page, so it adjusts better to the new UI design
-Adds a generic table for multiple purposes
-Componentized things like the chainaddresses, subdomains and luminochannels, to be used in other places of Rif sections

DomainDetailsResized

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -192,6 +192,7 @@
     "react-router-dom": "^4.2.2",
     "react-select": "^1.0.0",
     "react-simple-file-input": "^2.0.0",
+    "react-table": "7.1.0",
```
Comment (marcosmartinez7) [inline package.json]: We should not use a specific version at least there is some reason for that

File: ui/app/rif/components/table/genericTable.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { useTable, usePagination } from 'react-table';
+import Pagination from './pagination';
+
+const Table = ({ title, columns, data, pageSize, displayColumnHeader, className }) => {
```
Comment (marcosmartinez7) [inline ui/app/rif/components/table/genericTable.js]: Like this !

File: ui/app/rif/components/table/pagination.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    const visiblePages = this.getVisiblePages(page, this.props.pages);
+
+    this.setState({
+      visiblePages: this.filterPages(visiblePages, this.props.pages),
+    });
+
+    this.props.onPageChange(page - 1);
+  }
+
+  getVisiblePages = (page, total) => {
```
Comment (marcosmartinez7) [inline ui/app/rif/components/table/pagination.js:59]: Why is this needed? isnt this handled by react table by default?

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js [hunk 4]
```diff
@@ -158,100 +160,156 @@ class DomainsDetailActiveScreen extends Component {
...
+          </div>
+          <div className={''}>
+            <span>{chainAddress.address}</span>
+          </div>
+        </div>
+      tableRow.actions =
+        <div className={''}>
+          <FontAwesomeIcon
+            icon={faPen}
+            // color={'#FFFFFFFFFFF'}
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js]: Remove if unused

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js [hunk 5]
```diff
@@ -158,100 +160,156 @@ class DomainsDetailActiveScreen extends Component {
...
+            onClick={() => {}}
+          />
+          <FontAwesomeIcon
+            icon={faTimes}
+            // color={'#FFFFFFFFFFF'}
+            className={''}
+            onClick={() => {}}
+          />
+        </div>
+      data.push(tableRow);
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js]: Remove if unused

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js [hunk 6]
```diff
@@ -158,100 +160,156 @@ class DomainsDetailActiveScreen extends Component {
...
+        accessor: 'content',
+      },
+      {
+        Header: 'actions',
+        accessor: 'actions',
+      },
+    ];
+
+    // TODO Fede
+    // Acá van los styles para la tabla
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js]: Lets use english for comments

Comment (marcosmartinez7) [reply ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js]: and also github names for TODOs

**Review by marcosmartinez7 — CHANGES_REQUESTED** [review]

File: ui/app/rif/components/table/pagination.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const defaultButton = props => <button {...props}>{props.children}</button>;
+
+export default class Pagination extends Component {
```
Comment (marcosmartinez7) [inline ui/app/rif/components/table/pagination.js:6]: Does we really need a custom pagination component? why?

File: package.json (hunk 1 above)
Comment (rorolopetegui) [reply package.json]: Not a reason in particular, there i updated it to be always the latest version

File: ui/app/rif/components/table/pagination.js (hunk 3 above)
Comment (rorolopetegui) [reply ui/app/rif/components/table/pagination.js:59]: You need to send a pagination component for the react table, you can use it to change the pages. All the examples in react-table doc was using something like this. You can set the page size, and other things, but there's no way to get the visible pages

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js (hunk 4 above)
Comment (rorolopetegui) [reply ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js]: I was going to let this color att here, so @fmelo can use this. But he can also assign the color in CSS

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js (hunk 6 above)
Comment (rorolopetegui) [reply ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActive.js]: Ok, this was changed in last commit

File: ui/app/rif/components/table/pagination.js (hunk 3 above)
Comment (JONAF2103) [reply ui/app/rif/components/table/pagination.js:59]: I don't know about this, seems to me that maybe those were examples because react table should resolve this for you instead of you making the pagination by hand, it looks weird

**Review by JONAF2103 — CHANGES_REQUESTED** [review]

**Review by marcosmartinez7 — APPROVED** [review]

File: ui/app/rif/components/table/pagination.js (hunk 3 above)
Comment (rorolopetegui) [reply ui/app/rif/components/table/pagination.js:59]: Yesterday i talked to @marcosmartinez7, this is for the custom pagination, React-Table only manages functions like goToPage, and an index for the pages, if you want to code something like we need in the Figma design (Page's indexes shown), we need to use this pagination component.

File: ui/app/rif/components/table/pagination.js (hunk 7 above)
Comment (rorolopetegui) [reply ui/app/rif/components/table/pagination.js:6]: As we spoke yesterday, in resume, React-table doesnt manage the custom pagination that we need. With Page's indexes shown.

File: old-ui/app/css/index.css [hunk 8]
```diff
@@ -654,7 +654,7 @@ input.large-input {
   overflow-x: hidden;
   /*overflow-y: auto;*/
   overflow-y: -moz-scrollbars-none;
-  max-height: 585px;
+  max-height: 568px;
```
Comment (JONAF2103) [inline old-ui/app/css/index.css:657]: You shouldn't put a fixed value for this, instead of pixels we should be using em, rem or pt instead

File: ui/app/rif/components/chainAddresses.js [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+      this.setState({chainAddresses: chainAddresses});
+    }
+  }
+
+  convertChainAddressesToTableData () {
+    const { isOwner, classes } = this.props;
+    return this.state.chainAddresses.map((chainAddress) => {
+      const address = getChainAddressByChainAddress(chainAddress.chain);
+      const icon = address.icon ? address.icon : DEFAULT_ICON;
+      const item = <ItemWithActions contentClasses={classes.content} actionClasses={classes.contentActions} enableEdit={isOwner} enableDelete={isOwner} text={chainAddress.address} leftIcon={icon} onDeleteClick={this.onDeleteClick.bind(this, chainAddress.chain)}>
```
Comment (JONAF2103) [inline ui/app/rif/components/chainAddresses.js]: Be careful with this, you should put the component declaration in jsx sintax, that means you need to put a open parenthesis and close parenthesis at the end instead.

File: ui/app/rif/components/subDomains.js [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+      .then(subdomains => {
+        this.setState({subdomains: subdomains});
+      });
+  }
+
+  getData () {
+  const { classes } = this.props;
+  if (this.state.subdomains) {
+      return this.state.subdomains.map((subdomain) => {
+        const item = <ItemWithActions contentClasses={classes.content} actionClasses={classes.contentActions} text={subdomain.name} enableRightChevron={true} />
```
Comment (JONAF2103) [inline ui/app/rif/components/subDomains.js]: Same thing here you should always use parenthesis for this.

File: ui/app/rif/components/table/genericTable.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    {
+      columns,
+      data,
+      initialState: { pageIndex: 0, pageSize: pageSize || 10 },
+    },
+    usePagination,
+  );
+  const styles = classes || {};
+  return (
+    <>
```
Comment (JONAF2103) [inline ui/app/rif/components/table/genericTable.js]: What's this <> can you explain?

File: ui/app/rif/components/tokenNetworkChannels.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  loadData () {
+    const { classes } = this.props;
+    this.getChannels().then(channels => {
+      channels.map((channelJson) => {
+        const channel = channelJson[Object.keys(channelJson)[0]];
+        const tableRow = {};
+        tableRow.content =
+          <div className={classes.content}>
```
Comment (JONAF2103) [inline ui/app/rif/components/tokenNetworkChannels.js:32]: Same thing here, parenthesis is a good practice when you define jsx react components

File: ui/app/rif/components/customButton.js [hunk 13]
```diff
@@ -4,25 +4,31 @@ import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
...
+      return (
+        <div id="buttonNew" className={className.button} onClick={onClick}>
+          <div id="buttonBody">
+            {fontAwesomeIcon &&
+              <FontAwesomeIcon icon={fontAwesomeIcon} className={className.icon}/>
+            }
+            {svgIcon &&
+              svgIcon
+            }
+            {(text && text !== '') &&
```
Comment (marcosmartinez7) [inline ui/app/rif/components/customButton.js]: This check can be done like this:

`(text)  &&`

File: ui/app/rif/pages/common-style-pages.css [hunk 14]
```diff
@@ -15,10 +15,11 @@
 .hand-over {
   cursor: pointer;
 }
+
 .body {
   width: 100%;
-  overflow: scroll;
-  height: 26.5em;
+  overflow-y: auto;
+  height: 352px;
```
Comment (JONAF2103) [inline ui/app/rif/pages/common-style-pages.css:22]: Why is this fixed in pixels, can we change it to be not a fixed value?

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewSubdomain/index.js [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  constructor(props) {
+    super(props);
+    this.state = {
+      newSubdomain: {},
+    };
+  }
+
+  async addSubdomain () {
+    /**
```
Comment (JONAF2103) [inline ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewSubdomain/index.js]: If you fixed this you could remove the TODO comment.

File: ui/app/rif/components/luminoNetworkChannels.js [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+          </div>
+      }
+      {
+        data.length === 0 &&
+        <div>
+          <span>Lumino Channels</span>
+          <span>No channels found</span>
+        </div>
+      }
+        {(isOwner || false) &&
```
Comment (marcosmartinez7) [inline ui/app/rif/components/luminoNetworkChannels.js]: I dont understand the condition, why is the right side needed? 

Why dont just check  ` {(isOwner) &&` ?

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewTokenNetworkAddress/addNewChainAddressToResolver.js [hunk 17]
```diff
@@ -22,23 +23,22 @@ class AddNewChainAddressToResolver extends Component {
...
-		this.props.updateAddress(e.target.value);
+    this.props.updateAddress(e.target.value);
 	}
 
 	render () {
     const selectValue = ({value}) => {
       const icon = value.icon ? value.icon : DEFAULT_ICON;
       return (
-        <div className={'add-new-multicrypto-select-value'}>
+        <div className={''}>
```
Comment (JONAF2103) [inline ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewTokenNetworkAddress/addNewChainAddressToResolver.js]: Why is this class removed and why we have an empty string now, empty string it's the same as to not use the className attribute

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewSubdomain/index.js [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+     * TODO Rodrigo
+     * When confirming and submiting is redirecting to wrong direction
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewSubdomain/index.js]: Is this TODO fixed?

File: ui/app/rif/pages/domainsPage/domainsPage.js [hunk 19]
```diff
@@ -67,17 +67,17 @@ class DomainsScreen extends Component {
...
       </span>
-    </div>
+    </li>
   }
 
   render () {
     if (this.props.domains && this.props.domains.length > 0) {
       return (
-        <div className={'domains-list'}>
+        <ul className={'domains-list'}>
```
Comment (JONAF2103) [inline ui/app/rif/pages/domainsPage/domainsPage.js]: You can use just literal declaration instead.

File: ui/app/rif/pages/index.js [hunk 20]
```diff
@@ -53,7 +54,7 @@ function buildTabs (screenName, tabOptions) {
       index,
       title: tabDefinition.title,
       component: (
-        <div>
+        <div className="rif-tab-content__inside">
```
Comment (JONAF2103) [inline ui/app/rif/pages/index.js:57]: Good use of literals here

**Review by marcosmartinez7 — CHANGES_REQUESTED** [review]
Looks good, just some minor changes needed

**Review by JONAF2103 — CHANGES_REQUESTED** [review]

File: ui/app/rif/components/chainAddresses.js (hunk 9 above)
Comment (marcosmartinez7) [reply ui/app/rif/components/chainAddresses.js]: I think thats depends on how you use it later. 

If you use `data={item} `is ok.

File: ui/app/rif/components/luminoNetworkChannels.js [hunk 21]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  componentDidMount () {
+    this.loadChannels();
+  }
+
+  loadChannels () {
+    this.props.getChannelsGroupedByNetwork().then(networkChannels => {
+      const arrayNetworks = [];
+      for (let key of Object.keys(networkChannels)) {
```
Comment (JONAF2103) [inline ui/app/rif/components/luminoNetworkChannels.js]: Variable key doesn't change, it can be a const

File: old-ui/app/css/index.css (hunk 8 above)
Comment (rorolopetegui) [reply old-ui/app/css/index.css:657]: This was a change made by @fmelo

File: ui/app/rif/components/table/genericTable.js (hunk 11 above)
Comment (rorolopetegui) [reply ui/app/rif/components/table/genericTable.js]: This is just to wrap something, is the same as an empty <div>

File: ui/app/rif/components/subDomains.js (hunk 10 above)
Comment (rorolopetegui) [reply ui/app/rif/components/subDomains.js]: Yeah, you can do this, the jsx sintax will take where you open the tag, and finish it. <tag></tag>... but we can use the parenthesis as a convention. I changed this in next commit

File: ui/app/rif/pages/common-style-pages.css (hunk 14 above)
Comment (rorolopetegui) [reply ui/app/rif/pages/common-style-pages.css:22]: I dont really remember this change, in this PR i didn't touch any css... maybe because i merge it with other branches

Comment (rorolopetegui) [reply ui/app/rif/pages/common-style-pages.css:22]: Remember that this PR has changes of @fmelo

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewTokenNetworkAddress/addNewChainAddressToResolver.js (hunk 17 above)
Comment (rorolopetegui) [reply ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewTokenNetworkAddress/addNewChainAddressToResolver.js]: This will be empty, cause @fmelo needs to put some styling here

File: ui/app/rif/components/table/genericTable.js (hunk 11 above)
Comment (rorolopetegui) [reply ui/app/rif/components/table/genericTable.js]: There i changed it to be <div> instead of <>

File: ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewTokenNetworkAddress/addNewChainAddressToResolver.js (hunk 17 above)
Comment (rorolopetegui) [reply ui/app/rif/pages/domainsDetailPage/domainDetailActive/addNewTokenNetworkAddress/addNewChainAddressToResolver.js]: I deleted all classnames empty

File: ui/app/rif/pages/domainsPage/domainsPage.js (hunk 19 above)
Comment (rorolopetegui) [reply ui/app/rif/pages/domainsPage/domainsPage.js]: There i changed it

File: ui/app/rif/components/luminoNetworkChannels.js (hunk 21 above)
Comment (rorolopetegui) [reply ui/app/rif/components/luminoNetworkChannels.js]: Changed to be const

**Review by JONAF2103 — APPROVED** [review]
LGTM

**Review by marcosmartinez7 — APPROVED** [review]
